tensorflow/tsl/platform/windows/env.cc

```diff
@@ -53,17 +53,30 @@ class StdThread : public Thread {
  public:
   // thread_options is ignored.
   StdThread(const ThreadOptions& thread_options, const string& name,
-            std::function<void()> fn)
```

Comment:
Are these changes necessary for this PR? It would be easier for me to review without them, I'm not a Windows expert.

Reply:
Yes, they're needed, when `std::thread` is detached it's not possible to get the id of the thread like it's currently done in the destructor `std::thread::id thread_id = thread_.get_id();` and then remove the entry from the map `GetThreadNameRegistry().erase(thread_id);`

I think it's better to make it similar to the POSIX variant, when adding/removing from the map is done in a separate function that encapsulates a call to the `fn`